File: app/src/main/java/net/cynreub/subly/ui/home/HomeUiState.kt
package net.cynreub.subly.ui.home

import net.cynreub.subly.domain.model.Subscription
import net.cynreub.subly.domain.usecase.SubscriptionStats

data class HomeUiState(
    val upcomingSubscriptions: List<Subscription> = emptyList(),
    val stats: SubscriptionStats = SubscriptionStats(
        totalMonthly = 0.0,
        totalYearly = 0.0,
        activeCount = 0,
        categoryBreakdown = emptyMap()
    ),
    val isLoading: Boolean = true,
    val error: String? = null
)

File: subly_ios/Subly/Subly/UI/Home/HomeUiState.swift
import Foundation

struct HomeUiState {
    var upcomingSubscriptions: [Subscription] = []
    var stats: SubscriptionStats = SubscriptionStats(
        totalMonthly: 0,
        totalYearly: 0,
        activeCount: 0,
        categoryBreakdown: [:]
    )
    var categorySpend: [CategorySpend] = []
    var isLoading: Bool = true
    var error: String? = nil

    /// Top 3 subscriptions by monthly cost equivalent
    var topSubscriptions: [Subscription] { Array(upcomingSubscriptions.sorted { $0.monthlyAmount > $1.monthlyAmount }.prefix(3)) }
}
